src/lib/Repository/UiLocationService.php

```diff
@@ -46,20 +57,43 @@ public function loadLocations(ContentInfo $contentInfo)
     {
         $locations = $this->locationService->loadLocations($contentInfo);
 
-        $uiLocations = $this->buildUiLocations($locations, $contentInfo);
+        $uiLocations = $this->buildUiLocations($locations);
         $uiLocations = $this->prioritizeMainLocation($uiLocations);
 
         return $uiLocations;
     }
 
-    private function buildUiLocations(array $locations, ContentInfo $contentInfo)
+    /**
+     * Deletes secondary locations.
+     * Throws exception if there is an attempt to delete the main location.
+     *
+     * @param array $locationIds
+     *
+     * @throws ForbiddenException
+     */
+    public function deleteSecondaryLocations(array $locationIds)
+    {
+        foreach ($locationIds as $locationId) {
+            $location = $this->locationService->loadLocation($locationId);
+            $uiLocation = new UiLocation($location);
```

Comment:
I wouldn't create an incomplete `UiLocation` here. Writing `if ($location->id === $location->contentInfo->mainLocationId)` isn't more complicated.

If you'd rather avoid the code duplication, then you could:
- create a protected method `UiLocationService::isMain(Location $location)`, that does this comparison, and return a boolean
- use that method in `buildUiLocation()`
- use that method here